tests/test/Foundry.t.sol

```diff
@@ -27,4 +27,28 @@ contract FoundryTest is Test {
         counter2.set(x);
         assertEq(counter2.cnt(), x);
     }
+
+    function testEtchConcrete() public {
+        vm.etch(address(0x42), hex"60425f526001601ff3");
+        (bool success, bytes memory retval) = address(0x42).call("");
+
+        assertTrue(success);
+        assertEq(retval.length, 1);
+        assertEq(uint256(uint8(retval[0])), 0x42);
+    }
+
+    function testEtchSymbolicAddr(address who) public {
+        vm.etch(who, hex"60425f526001601ff3");
+        (bool success, bytes memory retval) = who.call("");
+
+        assertTrue(success);
+        assertEq(retval.length, 1);
+        assertEq(uint256(uint8(retval[0])), 0x42);
```

Comment:
suppose you create a new contract `c` at the beginning of this function, and add another assertion here saying that the code of `c` is unchanged, then it shouldn't pass because `who` may be equal to `c`, but the current version will not catch that, as i mentioned earlier.